deployment/common/changeset/test_helpers.go

```diff
@@ -33,7 +33,7 @@ func WrapChangeSet[C any](fn deployment.ChangeSet[C]) func(e deployment.Environm
 }
 
 // ApplyChangesets applies the changeset applications to the environment and returns the updated environment.
-func ApplyChangesets(t *testing.T, e deployment.Environment, timelocksPerChain map[uint64]*gethwrappers.RBACTimelock, changesetApplications []ChangesetApplication) (deployment.Environment, error) {
+func ApplyChangesets(t *testing.T, e deployment.Environment, timelocksPerChain map[uint64]*gethwrappers.RBACTimelock, callProxiesPerChain map[uint64]*gethwrappers.CallProxy, changesetApplications []ChangesetApplication) (deployment.Environment, error) {
```

Comment:
let's do map[uint64]*ProposalContracts

```
type ProposalContracts struct {
*gethwrappers.RBACTimelock
*gethwrappers.CallProxy
}
```

it will be easier for callers b/c products already track contracts like map[selector]ProductContracts

and more importantly it seems never valid to skew the keys in the maps so less error prone

and it's more extensible if it ever changes again